src/value.rs

```diff
@@ -0,0 +1,763 @@
+                let v = i16::deserialize(typ, val)?;
+                Ok(PyInt::new(py, v).into_any())
+            }
+            NativeType::Int => {
+                let v = i32::deserialize(typ, val)?;
+                Ok(PyInt::new(py, v).into_any())
+            }
+            NativeType::BigInt => {
+                let v = i64::deserialize(typ, val)?;
+                Ok(PyInt::new(py, v).into_any())
+            }
+            NativeType::Varint => {
+                let v = CqlVarint::deserialize(typ, val)?;
+                let big: num_bigint_04::BigInt = v.into();
+                call_python_constructor(py, "builtins", "int", (big.to_string(),), None)
+                    .map_err(DeserializationError::new)
+            }
+            NativeType::Double => {
+                let v = f64::deserialize(typ, val)?;
+                Ok(PyFloat::new(py, v).into_any())
+            }
+            NativeType::Float => {
+                let v = f32::deserialize(typ, val)?;
+                Ok(PyFloat::new(py, v as f64).into_any())
+            }
+            NativeType::Ascii | NativeType::Text => {
+                let v = <&str as DeserializeValue<'frame, 'metadata>>::deserialize(typ, val)?;
+                Ok(PyString::new(py, v).into_any())
+            }
+            NativeType::Boolean => {
+                let v = bool::deserialize(typ, val)?;
+                Ok(PyBool::new(py, v).to_owned().into_any())
+            }
+            NativeType::Date => {
+                let v = CqlDate::deserialize(typ, val)?;
+                let date: chrono_04::NaiveDate = v.try_into().map_err(DeserializationError::new)?;
+
+                py_call_method(
+                    py,
+                    "datetime",
+                    "date",
+                    "fromisoformat",
+                    (date.to_string(),),
+                    None,
+                )
+                .map_err(DeserializationError::new)
+            }
+            NativeType::Timestamp => {
+                let t = CqlTimestamp::deserialize(typ, val)?;
+                let millis = t.0;
+                let seconds = (millis as f64) / 1000.0;
+
+                let datetime_mod = py.import("datetime").map_err(DeserializationError::new)?;
+                let tz = datetime_mod
+                    .getattr("timezone")
+                    .map_err(DeserializationError::new)?
+                    .getattr("utc")
+                    .map_err(DeserializationError::new)?;
+
+                let kwargs = PyDict::new(py);
+                kwargs
+                    .set_item("tz", tz)
+                    .map_err(DeserializationError::new)?;
+
+                py_call_method(
+                    py,
+                    "datetime",
+                    "datetime",
+                    "fromtimestamp",
+                    (seconds,),
+                    Some(&kwargs),
+                )
+                .map_err(DeserializationError::new)
+            }
```

Comment:
This also look unnecessarily complicated. PyO3 has support for chrono and time - can't we use one of those here?